src/Mod/PartDesign/Gui/TaskRevolutionParameters.cpp

```
@@ -109,17 +114,15 @@ void TaskRevolutionParameters::fillAxisCombo(bool forceRefill)
     if (axesInList.empty())
         forceRefill = true;//not filled yet, full refill
 
-    if (forceRefill){
+    if (forceRefill) {
         ui->axis->clear();
-
-        for(size_t i = 0; i < axesInList.size(); i++){
-            delete axesInList[i];
-        }
-        this->axesInList.clear();
+        axesInList.clear();
 
         //add sketch axes
-        PartDesign::ProfileBased* pcFeat = static_cast<PartDesign::ProfileBased*>(vp->getObject());
-        Part::Part2DObject* pcSketch = dynamic_cast<Part::Part2DObject*>(pcFeat->Profile.getValue());
+        auto *pcFeat = dynamic_cast<PartDesign::ProfileBased*>(vp->getObject());
+        if (!pcFeat)
+            return;

```

Same here. Do not silently return from the constructor.